### PR TITLE
Feature - Advance Search

### PR DESCRIPTION
**Pull request by sadmananik**
* Added nationId parameter on searchEvent GQL query

* Update new gateway types and added nationId in searchEvents queries on client

* Added nationalId query variable in SEARCH_EVENTS tests

* Refactor fatherIdentifier & motherIdentifier on birthEventHandler

* Added informantIdentifier & deceasedIdentifier for birth and death declaration on elastic

* Added nationalId on queryBuilder

* Fixed elasticsearch unit test for queryBuilder

* ocrvs-4105 frontend (#4150)

* WIP: remove depecrated code and add nationalID

* WIP: comment out unrecognise property

* ocrvs-3756 Introduced Iconography on storybook using react-feather (#4055)

* Added new stories for Box and Trash Icon

* Introduced react-feather

* Added new icon template and introduced icon gallery using react-feather

* Removed unsued icon components

* Modify icon component to support all icons and update stories

* Add all custom icons

* Fix currentColor not being in 'colors'

* Rename form icons and icons to custom icons

* Fix iconography and change the copy a bit

* Upload versions of icons that have viewbox 1:1

* deprecated old components, updated copy

Co-authored-by: naftis <pyry.rouvila@gmail.com>
Co-authored-by: jpye-finch <jonathan@pye-finch.co.uk>

* add navigation

* advanced search page

* Update route

* Added navigationList props on Searchtool component

* Added License text for Icon component types file

* Update nationalId icon

* remove unused code

Co-authored-by: modupeadeonojobi <modupeadeonojobi@gmail.com>
Co-authored-by: naftis <pyry.rouvila@gmail.com>
Co-authored-by: jpye-finch <jonathan@pye-finch.co.uk>
Co-authored-by: corneliusifoghale <collinscornelius9@gmail.com>

* Replace searchType icons with new Icon component

* Fix searchEvents queries on client

Co-authored-by: modupeadeonojobi <modupeadeonojobi@gmail.com>
Co-authored-by: naftis <pyry.rouvila@gmail.com>
Co-authored-by: jpye-finch <jonathan@pye-finch.co.uk>
Co-authored-by: corneliusifoghale <collinscornelius9@gmail.com>

### CODE REVIEW

File: packages/components/src/SearchTool/SearchTool.tsx [hunk 1]
```diff
@@ -155,11 +163,17 @@ const ClearTextIcon = styled((props) => <ClearText {...props} />)`
 export interface ISearchType {
   label: string
   value: string
-  icon: React.ReactNode
-  invertIcon: React.ReactNode
+  icon?: React.ReactNode
+  invertIcon?: React.ReactNode
```
Comment (sadmananik) [inline packages/components/src/SearchTool/SearchTool.tsx:167]:
```suggestion
  icon: React.ReactNode
  invertIcon: React.ReactNode
```

File: packages/components/src/Accordion/Accordion.tsx [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+  name,
+  label,
+  options,
+  value,
+  nestedFields,
+  onChange
+}: IAccordionProps) => {
+  return (
+    <Container>
+      <h2>{label}</h2>
```
Comment (jpye-finch) [inline packages/components/src/Accordion/Accordion.tsx:132]: Should be using < Text /> here

File: packages/components/src/Accordion/Accordion.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import { Accordion, IAccordionProps } from './Accordion'
+import React from 'react'
+
+export default {
+  title: 'Data/Accordion',
+  component: Accordion,
+  parameters: {
+    docs: {
+      description: {
+        component: `\`<Accordion />\` `
```
Comment (naftis) [inline packages/components/src/Accordion/Accordion.stories.tsx:22]: Descripting the component would be nice. If you need inspiration, you can see what kind of text [Gov.uk](https://design-system.service.gov.uk/components/accordion/) or [MUI](https://mui.com/material-ui/react-accordion/) use here.

File: packages/components/src/Accordion/Accordion.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+      }
+    }
+  }
+} as ComponentMeta<typeof Accordion>
+
+const Template: Story<IAccordionProps> = (args) => <Accordion {...args} />
```
Comment (naftis) [inline packages/components/src/Accordion/Accordion.stories.tsx:28]: Could the accordion have some content in the example?

File: packages/components/src/Accordion/Accordion.tsx [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+    display: inline-block;
+    height: 8px;
+    position: relative;
+    top: 7px;
+    vertical-align: top;
+    width: 8px;
+    left: 0;
+    margin-right: 10px;
+    transform: rotate(45deg);
+  }
```
Comment (naftis) [inline packages/components/src/Accordion/Accordion.tsx:34]: Could we use chevron `<Icon />` and maybe hide this browser default one?

Comment (jpye-finch) [reply packages/components/src/Accordion/Accordion.tsx:34]: we don't need the rotate animation also

**Review by Zangetsu101 — APPROVED** [review]

**Pull request merged**